storagenode/collector/service.go

```diff
@@ -1,33 +1,96 @@
 // Copyright (C) 2019 Storj Labs, Inc.
 // See LICENSE for copying information.
 
+// Package collector implements expired piece deletion from storage node.
 package collector
 
 import (
+	"context"
+	"errors"
 	"time"
 
 	"go.uber.org/zap"
+	monkit "gopkg.in/spacemonkeygo/monkit.v2"
 
+	"storj.io/storj/internal/sync2"
 	"storj.io/storj/storagenode/pieces"
 )
 
+var mon = monkit.Package()
+
 // Config defines parameters for storage node Collector.
 type Config struct {
-	Interval time.Duration
+	Interval time.Duration `help:"how frequently expired pieces are collected" default:"1h0m0s"`
 }
 
 // Service implements collecting expired pieces on the storage node.
 type Service struct {
 	log        *zap.Logger
 	pieces     *pieces.Store
 	pieceinfos pieces.DB
+
+	Loop sync2.Cycle
 }
 
 // NewService creates a new collector service.
-func NewService(log *zap.Logger, pieces *pieces.Store, pieceinfos pieces.DB) *Service {
+func NewService(log *zap.Logger, pieces *pieces.Store, pieceinfos pieces.DB, config Config) *Service {
 	return &Service{
 		log:        log,
 		pieces:     pieces,
 		pieceinfos: pieceinfos,
+		Loop:       *sync2.NewCycle(config.Interval),
 	}
 }
+
+// Run runs monitor service
+func (service *Service) Run(ctx context.Context) (err error) {
+	defer mon.Task()(&ctx)(&err)
+
+	return service.Loop.Run(ctx, func(ctx context.Context) error {
+		err := service.collect(ctx)
+		if err != nil {
+			service.log.Error("error during collecting pieces: ", zap.Error(err))
+		}
+		return nil
+	})
+}
+
+// Close stops the collector service.
+func (service *Service) Close() (err error) {
+	service.Loop.Close()
+	return nil
+}
+
+func (service *Service) collect(ctx context.Context) (err error) {
+	defer mon.Task()(&ctx)(&err)
+
+	now := time.Now()
+	const maxBatches = 100
+	const batchSize = 1024
```

Comment:
Why 1024, and not 1000? We have already hardcoded limits to 1000 in several places.